src/utilities/googleAnalytics.ts

```diff
@@ -38,3 +47,53 @@ export const getNotionalActionFromHintState = (hint: SwapFormSubmitButtonHintSta
             return "";
     }
 }
+
+export const pushEvent = (eventName: string, eventValue: string | number, sessionId?: string, amm?: AugmentedAMM, agent?: string) => {
+    // with context
+    if (amm) {
+        const pool = amm.protocol + (amm.rateOracle.protocolId === 5 || amm.rateOracle.protocolId === 6 ? "_borrow" : "" );
+        window.dataLayer.push({'event': eventName, 'value': eventValue, 'sessionId': sessionId, 'pool': pool, 'agent': agent});
+    } else { // without context
+        window.dataLayer.push({'event': eventName, 'value': eventValue, 'sessionId': sessionId});
+    }
+}
+
+export const pushTxSubmition = (
```

Comment:
maybe check if window.dataLayer is present at the top of the function before actually using it.

something like

if(!window.dataLayer) {
  return;
}

Reply:
[checked](https://github.com/Voltz-Protocol/voltz-ui/pull/268/commits/9bb0958572bdde864d23200e7f12fd6c19910057)